library/dokku_network_property.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# from ansible.module_utils.basic import *
+from ansible.module_utils.basic import AnsibleModule
+import subprocess
+
+DOCUMENTATION = """
+---
+module: dokku_network_property
+short_description: Set or clear a network property for a given dokku application
+options:
+  app:
+    description:
+      - The name of the app
+    required: True
+    default: null
+    aliases: []
+  network:
+    description:
+      - The name of the network to attach to
+    required: True
+    default: null
+    aliases: []
+  property:
+    description:
+      - The attach network property
+    required: True
+    default: null
+    choices: [ "initial-network", "attach-post-create", "attach-post-deploy" ]
+    aliases: []
+author: Philipp Sessler
+requirements: [ ]
+"""
+
+EXAMPLES = """
+- name: Associates a network after a container is created but before it is started
+  dokku_network_property:
+    app: hello-world
+    network: example-network
+    property: attach-post-create
+
+- name: Associates the network at container creation
+  dokku_network_property:
+    app: hello-world
+    network: example-network
+    property: initial-network
+
+- name: Setting a global network property
+  dokku_network_property:
+    app: --global
+    network: example-network
+    property: attach-post-create
+
+- name: De-associate the container with the network.
+  dokku_network_property:
+    app: hello-world
+    network: example-network
+"""
+
+
+def dokku_network_property_set(data):
+    is_error = True
+    has_changed = False
+    meta = {"present": False}
+
+    command = "dokku network:set {0} {1} {2}".format(
+        data["app"],
+        data["property"] if "property" in data else "",
+        data["network"],
+    )
+
+    try:
+        subprocess.check_call(command, shell=True)
+        is_error = False
+        has_changed = True
+        meta["present"] = True
+    except subprocess.CalledProcessError as e:
+        meta["error"] = str(e)
+
+    return (is_error, has_changed, meta)
+
+
+def main():
+    fields = {
+        "app": {"required": True, "type": "str"},
+        "network": {"required": True, "type": "str"},
+        "property": {
+            "required": False,
+            "choices": ["initial-network", "attach-post-create", "attach-post-deploy"],
```

Comment:
Is there a reason to limit the set of properties to these three?

How about also supporting `static-web-listener` (not sure whether there are more)?properties

Reply:
After skimming through code & documentation, i think these properties are also supported:

- bind-all-interfaces
- static-web-listener
- tld

I'm gonna add them.

Reply:
Thanks!

I leave it up to you whether you want to keep the list of properties here explicitly (which means we need to maintain the list in case dokku changes) or whether we simply allow any value and bubble up the error message to the user.

Whatever you decide, it may make sense to mention the list you now collected somewhere in the documentation of the module

Reply:
Yes, I get your point. I think it makes to accept any string and let Dokku do the verification.